.github/actions/generate_manifest/action.yml

```diff
@@ -0,0 +1,33 @@
+name: "Generate Manifest"
+description: "Generate a manifest file for the OTA client"
+inputs:
+  dir:
+    description: "The directory containing the SquashFS and patches files"
+    required: true
+  output_manifest:
+    description: "The output manifest file"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: setup python
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.8
+
+    - name: install build deps
+      shell: bash
+      run: |
+        python -m pip install -U pip
+        python -m pip install -U jsonschema
```

Comment:
I recommend to use `pydantic` to define the manifest.

```python
import json
from datetime import datetime
from typing import List, Literal

from pydantic import BaseModel


class ReleasePackage(BaseModel):
    filename: str
    version: str
    type: Literal["squashfs", "patch"]
    architecture: Literal["x86_64", "arm64"]
    size: int
    checksum: str


class Manifest(BaseModel):
    schema_version: str = "1"
    date: datetime
    packages: List[ReleasePackage]

json_schema = json.dumps(Manifest.model_json_schema())
```

Reply:
Thank you, fixed.